build.gradle

```diff
@@ -162,41 +158,71 @@ tasks.create('testNaming') {
 	}
 }
 
-// Create a test task with given name, environments dir/file and is it development mode task
-void createTestTask(String name, String environments, boolean devMode) {
-	tasks.create(name) {
-		// Compile Skript and check test naming
+// Create a test task with given name, environments dir/file, dev mode and java version.
+void createTestTask(String name, String environments, boolean devMode, int javaVersion) {
+	tasks.register(name, JavaExec) {
 		dependsOn jar, testNaming
-		doFirst {
-			if (devMode) {
-				standardInput = System.in
-			}
-			javaexec {
-				group = 'execution'
-				classpath = files([
-					'build' + File.separator + 'libs' + File.separator + 'Skript.jar',
-					project.configurations.runtimeClasspath.find { it.name.startsWith('gson') },
-					sourceSets.main.runtimeClasspath
-				])
-				main = 'ch.njol.skript.tests.platform.PlatformMain'
-				args = [
-					'test_runners',
-					'src/test/skript/tests',
-					'src/test/resources/runner_data',
-					environments,
-					devMode
-				]
-			}
+		javaLauncher = javaToolchains.launcherFor {
+	        languageVersion = JavaLanguageVersion.of(javaVersion)
+	    }
+		if (devMode) {
+			standardInput = System.in
 		}
+		group = 'execution'
+		classpath = files([
+			'build' + File.separator + 'libs' + File.separator + 'Skript.jar',
+			project.configurations.runtimeClasspath.find { it.name.startsWith('gson') },
+			sourceSets.main.runtimeClasspath
+		])
+		main = 'ch.njol.skript.tests.platform.PlatformMain'
+		args = [
+			'test_runners',
+			'src/test/skript/tests',
+			'src/test/resources/runner_data',
+			environments,
+			devMode
+		]
 	}
 }
 
+def latestEnv = 'java17/paper-1.19.json'
+def latestJava = 17
+def oldestJava = 8
+
+tasks.withType(JavaCompile).configureEach {
+	options.compilerArgs += ["-source", "" + oldestJava, "-target", "" + oldestJava]
+}
+
 // Register different Skript testing tasks
-createTestTask('quickTest', 'src/test/skript/environments/main/paper-1.18.2.json', false)
-createTestTask('skriptTest', 'src/test/skript/environments/main', false)
-createTestTask('skriptTestFull', 'src/test/skript/environments/', false)
-createTestTask('skriptTestDev', 'src/test/skript/environments/main/' + (project.property('testEnv') == null
-	? 'paper-1.17.json' : project.property('testEnv') + '.json'), true)
+createTestTask('quickTest', 'src/test/skript/environments/' + latestEnv, false, latestJava)
+createTestTask('skriptTest', 'src/test/skript/environments/java17', false, latestJava)
```

Comment:
Do you think it'd be better to rename this test skriptTestJava17 (to keep in line with naming of skriptTestJava8)?
(possibly also rename skriptTestFull to skriptTest after that)

Reply:
Sure